server/src/main/java/org/elasticsearch/transport/TransportKeepAlive.java

```diff
@@ -48,28 +48,31 @@
 
     static final int PING_DATA_SIZE = -1;
 
+    private static final BytesReference PING_MESSAGE;
+
+    static {
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            out.writeByte((byte) 'E');
+            out.writeByte((byte) 'S');
+            out.writeInt(PING_DATA_SIZE);
+            PING_MESSAGE = out.copyBytes();
+        } catch (IOException e) {
+            throw new AssertionError(e.getMessage(), e); // won't happen
+        }
+    }
```

Comment:
What is the purpose of this block? Is it just to generate a `BytesReference` containing `"ES"`? If so, could we not just do:

```suggestion
    private static final BytesReference PING_MESSAGE = new BytesArray("ES");
```

Reply:
We also have the `int` `PING_DATA_SIZE`, but yea the only purpose of the block is setting up the message but I don't think we have a nicer way of building it right at this point.

Reply:
Oh, I missed the part where it writes the size as well.